src/doc/imageinput.rst

```diff
@@ -360,10 +362,11 @@ flexible functionality.  A few representative examples follow:
 
 * Flip an image vertically upon reading, by using *negative* ``y`` stride::
 
-    unsigned char pixels[spec.width * spec.height * spec.nchannels];
+    std::vector<unsigned char> pixels(spec.width * spec.height * spec.nchannels);
```

Comment:
Just as an FYI, if I were writing this for real, I think the best idiom would be:

    auto pixels = std::unique_ptr<unsigned char[]>(new unsigned char [spec.width * spec.height * spec.nchannels]);

That would be freed upon scope exit (like std::vector), but it's much more efficient -- new[] doesn't initialize the memory, but std::vector will fill it with zeroes unnecessarily!

Reply:
How about I change them to the following. Seem ok?
`auto pixels = std::make_unique<unsigned char[]>(spec.width * spec.height * spec.nchannels);`

Reply:
Will that really work without the `new`?

Reply:
Yes, make_unique does work like that, however it looks like the semantics of make_unique include value initialization. After trying it out on compiler explorer the make_unique calls will include some form of memset basically (rep stosq)... c++ why are you the way you are...  https://godbolt.org/z/9PEYKbY6f

I'll replace all of them with your suggestion then.